x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_list/credentials_list.test.tsx

```diff
@@ -0,0 +1,276 @@
+
+describe('Credentials', () => {
+  const apiToken: IApiToken = {
+    name: '',
+    type: ApiTokenTypes.Private,
+    read: true,
+    write: true,
+    access_all_engines: true,
+    key: 'abc-1234',
+  };
+
+  const mockKea = ({ values = {}, actions = {} }) => {
+    const mergedValues = {
+      apiTokens: [],
+      meta: {
+        page: {
+          current: 1,
+          size: 10,
+          total_pages: 1,
+          total_results: 1,
+        },
+      },
+      ...values,
+    };
+
+    const mergedActions = {
+      deleteApiKey: jest.fn(),
+      fetchCredentials: jest.fn(),
+      showCredentialsForm: jest.fn(),
+      ...actions,
+    };
+
+    (useValues as jest.Mock).mockImplementationOnce(() => mergedValues);
+    (useActions as jest.Mock).mockImplementationOnce(() => mergedActions);
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders', () => {
+    mockKea({});
+    const wrapper = shallow(<CredentialsList />);
+    expect(wrapper.find(EuiBasicTable)).toHaveLength(1);
+  });
+
+  describe('items', () => {
+    it('sorts items by id', () => {
+      mockKea({
+        values: {
+          apiTokens: [
+            {
+              ...apiToken,
+              id: 2,
+            },
+            {
+              ...apiToken,
+              id: undefined,
+            },
+            {
+              ...apiToken,
+              id: 1,
+            },
+          ],
+        },
+      });
+      const wrapper = shallow(<CredentialsList />);
+      const { items } = wrapper.find(EuiBasicTable).props();
+      expect(items.map((i: IApiToken) => i.id)).toEqual([undefined, 1, 2]);
+    });
+  });
+
+  describe('pagination', () => {
+    it('derives pagination from meta object', () => {
+      mockKea({
+        values: {
+          meta: {
+            page: {
+              current: 6,
+              size: 55,
+              total_pages: 1,
+              total_results: 1004,
+            },
+          },
+        },
+      });
+
+      const wrapper = shallow(<CredentialsList />);
+      const { pagination } = wrapper.find(EuiBasicTable).props();
+      expect(pagination).toEqual({
+        pageIndex: 5,
+        pageSize: 55,
+        totalItemCount: 1004,
+        hidePerPageOptions: true,
+      });
+    });
+
+    it('will default pagination values if `page` is not available', () => {
+      mockKea({
+        values: {
+          meta: {},
+        },
+      });
+
+      const wrapper = shallow(<CredentialsList />);
+      const { pagination } = wrapper.find(EuiBasicTable).props();
+      expect(pagination).toEqual({
+        pageIndex: 0,
+        pageSize: 0,
+        totalItemCount: 0,
+        hidePerPageOptions: true,
+      });
+    });
+  });
+
+  describe('columns', () => {
+    let columns: Array<{
+      render: (token: object) => any;
+      actions: any;
+    }>;
+    const showCredentialsForm = jest.fn();
+    const deleteApiKey = jest.fn();
+
+    beforeEach(() => {
```

Comment:
Since `columns` isn't changing depending on props, we can pretty safely just do a single `beforeAll` instead of `beforeEach` for some #microperf

```suggestion
    beforeAll(() => {
```